mirascope/core/base/tool.py

```diff
@@ -84,6 +85,17 @@ def call(self) -> str:
     __custom_name__: ClassVar[str] = ""
     tool_config: ClassVar[ToolConfig] = ToolConfig()
     model_config = ConfigDict(arbitrary_types_allowed=True)
+    _delta: str | None = None
+
+    @property
+    def delta(self) -> str | None:
+        return self._delta
```

Comment:
nit: have this return `None` and remove `_delta`

then in the partial tool construction overwrite the entire property with a lambda to return the delta (so that `_delta` doesn't show up in autocomplete)

Reply:
The pydantic structure does not allow for property overrides.
```python
/Users/koudai/PycharmProjects/mirascope/.venv/bin/python /Users/koudai/Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_119.py 
sagemaker.config INFO - Not applying SDK defaults from location: /Library/Application Support/sagemaker/config.yaml
sagemaker.config INFO - Not applying SDK defaults from location: /Users/koudai/Library/Application Support/sagemaker/config.yaml
Traceback (most recent call last):
  File "/Users/koudai/Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_119.py", line 11, in <module>
    for chunk, tool in stream:
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/openai/stream.py", line 76, in __iter__
    for chunk, tool in super().__iter__():
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/base/stream.py", line 146, in __iter__
    for chunk, tool in self.stream:
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/base/stream.py", line 432, in generator
    yield from handle_stream(
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/openai/_utils/_handle_stream.py", line 99, in handle_stream
    tool, current_tool_call, current_tool_type = _handle_chunk(
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/openai/_utils/_handle_stream.py", line 73, in _handle_chunk
    partial_tool.delta = cast(str, tool_call.function.arguments)
  File "/Users/koudai/PycharmProjects/mirascope/.venv/lib/python3.10/site-packages/pydantic/main.py", line 918, in __setattr__
    attr.__set__(self, value)
AttributeError: can't set attribute 'delta'

Process finished with exit code 1
```

Reply:
Ah ok, what if we just create a field `delta` instead of a property and exclude it from the tool schema generation (like we do for the `tool_call` attribute)?